src/Datasource/EntityTrait.php

```diff
@@ -787,6 +787,43 @@ public function getDirty()
     }
 
     /**
+     * return value if property exists otherwise throw Exception
+     *
+     * @param string $property The field to get value for.
+     * @return mixed
+     * @throws \InvalidArgumentException if the property name doesn't exist
+     */
+    public function &getOrFail($property)
+    {
+        $value = $this->get($property);
+        if (!$this->propertyExists($property) && $value === null) {
+            throw new \InvalidArgumentException('Property "' . $property . '" not found.');
+        }
+
+        return $value;
+    }
+
+    /**
+     * Check if real or virtual property for existence
+     *
+     * @param string $property The field to check for existence.
+     * @return bool
+     */
+    public function propertyExists($property)
```

Comment:
How is this better than `isset($thing->property)` which will check virtual and concrete properties?

Reply:
isset() returns true if the value is null - but property could exist having the value NULL

Reply:
@mdlouhy Are you sure that `isset()` returns true for `null`?
http://php.net/manual/en/function.isset.php

Reply:
isset() will return false, if the property returns null (and probably? if the method/virtual property returns null), *but* we need to return true if the property or method (virtual property) exists, don't we?

Reply:
@robertpustulka your are right, of course - I mistyped ...
I meant: isset() returns FALSE if the property value is null - but I want to check if the property exists, so it should return true when value is set but null.
($entity->property = null; -> property exists)
It should ONLY return false if it really doesn't exist.
The reason is very simple. I have in my use cases some columns which can be null - for example a datetime column "deleted_date". In den beforeSave or afterSave I want to check if the value was changed to or is NULL without checking "isDirty()". In my routine I do or don't do some actions depending on the value available in the entity and is NULL.
I hope this make my intentions a little bit clearer.

Reply:
> ..without checking "isDirty()..

Why not? Checking whether an entity property has been changed is precisely that method is for.

Reply:
@ADmad isDirty feels wrong for me in that use case. I want to check if a property exists and I don't want to care if it was marked as dirty implicitly or explicitly. It feels better for me that way. I think that the code is easier to read that way.

Reply:
Can you show an e.g. of how want you want would be achieved using isDirty() vs the new method you are adding?